### DISCUSSION TITLE
Bug in reusing assignments

### DISCUSSION THREAD

**Discussion by sigged** [community/community#137250] [GitHub Education]
### Hi everyone,

We are trying to adapt our workflow to embrace the new assignment creation. Because we need multiple copies of the same assignment, we are considering "reuse assignment". 
However, this seems broken now. After selecting a different classroom and clicking "Reuse", we get the following error: **"Unexpected token '<', "<!doctype "... is not valid JSON"**

Update: reusing seems to work fine when selection a different organisation. But that won't do for our workflow.

<img width="349" alt="image" src="https://github.com/user-attachments/assets/f00cc789-cba6-4d86-a533-fd0b9aefc856">

**Comment by MiguelCardoso19** [comment]
Try clearing your browser cache or using a different browser. If that doesn’t help, you may need to manually duplicate the assignment until GitHub fixes the issue. If the problem persists, contacting GitHub Support is your best option.

**Reply by sigged** [reply]
Cleared cache and used a different browser.

Seems like the issue is caused by a 404. When clicking "Reuse", a GET request is issued to the future location of the cloned assignment, which, at this point does not exist yet. I will try GitHub support.

**Reply by Shakattack76** [reply]
Ty

**Comment by slovak-edhouse** [comment]
Hi all, I have also problem with assignment reuse in GitHub classroom. Reuse of assignments is not working, but I get different error: "JSON.parse: unexpected character at line 1 column 1 of the JSON data".

![image](https://github.com/user-attachments/assets/dba257a8-fb44-42d8-a908-2432e53ff5cf)

I try different browsers (FF and Chrome) with the same result.

**Comment by nimonian** [comment]
Exact same problem here, thought I was losing my mind as it was working recently. A bug must have been introduced by a recent update.

![image](https://github.com/user-attachments/assets/b658731d-f5ae-44c6-9ad8-df9c1f9f2137)

I agree with @sigged that the problem occurs because a GET request is being made to the (as yet uncreated) target assignment URL.

![image](https://github.com/user-attachments/assets/8526b5fb-4a1c-41ab-b32e-ca963c1186f0)

This is a pretty significant issue for us as we have a cohort starting very soon.

**Reply by Ujoa** [reply]
Agreed. Encountering this now after returning from Tuesday's fiasco of classrooms failing. Seems like website is still unstable and this is concerning.

**Reply by nimonian** [reply]
I found out what the issue is.

If you try to reuse an assignment into a classroom within the same organisation, it will fail.

If you reuse an assignment into a classroom in a different organisation, it will succeed.

It seems like Github are trying to force us to use "one classroom per org" which is not how we use it, and I imagine there are many people who have many classrooms per org and want to reuse assignments between them.

**Comment by beak85** [comment]
Same issue with one of my assignments. I then tried with another similar assignment and it worked fine. Went back to the first and still the same error. I wonder if there is something within the assignment repo that causes it. I used the same settings in each assignment except for the name and starting repo.

**Comment by Ujoa** [comment]
@github-staff this issue is still occurring.

**Comment by Ujoa** [comment]
After clearing cache and browser, as suggested above, the issue is still occurring. Here's a network log (hopefully this helps)
![image](https://github.com/user-attachments/assets/a4c1e0a6-3287-4e30-b88f-065b25d6edc5)

**Comment by nimonian** [comment]
I found out what the issue is.

If you try to reuse an assignment into a classroom within the same organisation, it will fail.

If you reuse an assignment into a classroom in a different organisation, it will succeed.

It seems like Github are trying to force us to use "one classroom per org" which is not how we use it, and I imagine there are many people who have many classrooms per org and want to reuse assignments between them.

**Reply by nimonian** [reply]
Oh it gets better

So the assignment has stalled at being created in the separate org

![image](https://github.com/user-attachments/assets/2960d4de-a4b7-464c-91b6-edc238a471bc)

And the base repo has been duplicated FIVE TIMES in the target organisation with a random id added to the slug

![image](https://github.com/user-attachments/assets/977357aa-d7c1-4759-8a7d-d42b3f415719)

I'm... at the point of giving up

**Reply by Ujoa** [reply]
yeah looks like assignment reuse is not feasible anymore. when it used to work.

**Comment by sierrajulietromeo** [comment]
Any news on this? Still not working as of yesterday.

**Comment by khammami** [comment]
I am experiencing the same issue when you try to reuse one assignment. It appears to be related to the new method GitHub Classroom uses to create assignments with starter code. The process requires a public template to create a copy under your organization. However, this copy is just a normal repository and not a template, and it can be private if you change the assignment's visibility. Therefore, when we use the 'reuse' feature, it likely checks if the starter code is a  template, causing it to fail.

update:

Here is what I get when I try to use the copied repo from a previous assignment in a new assignment
![Screenshot from 2024-10-11 09-30-14](https://github.com/user-attachments/assets/6b4a6a00-a204-4595-b921-cbaa97c5dd92)

**Reply by scmurdock** [reply]
You are correct! This solved our issue, thank you so much. We can't say how much time it will save our school.

**Reply by khammami** [reply]
I'm glad to hear that :)

**Reply by tturin1017** [reply]
I am new to github classrooms so my apologies for not understanding. Are you saying that in order for the reuse assignment feature to work, we must have our starter code be from a private template? I thought a starter template could not only be public. 
Thank you! @khammami

**Reply by joshuaguerin** [reply]
You are still helping people in 2026!

**Comment by saniales** [comment]
Just want to point out that this is still not working and same error keep appearing as everyone else here

**Comment by vindiarputra** [comment]


**Reply by tturin1017** [reply]
Hi, I believe I have seen this reponse  before. Unfortunately, I am not able to solve the issue with it.

**Comment by brantfaircloth** [comment]
The @github support for critical issues this like this one is woefully inadequate.  This bug continues to impact re-using assignments (something that is pretty critical to the operation of a classroom) with no suggestions/fixes from @github...

Similar to others, I am also stuck with the Unexpected token error when attempting to re-use an assignment.

**Comment by gublan24** [comment]
Hi guys. I am encountering the same error. 

You can create a temp classroom in a new organization to reuse the assignment there. Then, from the new classroom, you can reuse it back to the original classroom. 

I did it, still there some errors.

**Comment by bharlowRHC** [comment]
I admitted defeat and gave up. 

I have run into both issues here that @sigged and @nimonian have described. I do not have time to troubleshoot due to the fires that started a few hours after my GitHub Classroom duplicated a single assignment 26 times at the beginning of January 2025. Over a month later I finally found a few minutes to check it, and it says that the assignment still says it has not completed the copy at least it stopped at 26 duplications.

I am a big advocate of learning to use version control for my students. However, because of continued issues like this coupled with my newfound severe lack of time while dealing with recovery, my students will lose out on learning experiences such as GitHub Classroom assignments until I get more time to troubleshoot and deal with "workarounds".

**Comment by MrDGuy** [comment]
Here is a quick work-around from @khammami. Thank you for taking the time to figure this out.  When I go to the assignment repository, which has been named something like _org-name/class-name-assignment-name_, I then change this to a template and then I can re-use it for other classes of the same organization.

**Reply by kootsoop** [reply]
Thank-you! That seems to resolve it for me, too. When creating the initial assignments, the starter repository had to be a template... but the assignment creation seems to make a new repository that is NOT a template. Making the previous class's starter repositories templates manually has been fixing it for me.

**Reply by schrum2** [reply]
Agreed. As of today (July 3, 2025) the error still happens when you try to reuse an assignment. Note also that if you try to reuse the assignment from the overall classroom listing rather than the specific assignment, then the process fails silently rather than give the invalid JSON error above. However, by going into the starter repository actually tied to the assignment and making it a template, the reuse option worked.

Just to clarify for anyone still confused: the original repo that you made the assignment from and the starter repo for some past assignment you've used are actually different things. My original repo was already a template, but when you create a new assignment that uses such a repo as its template, another repo is made that is the actual starter repo for the assignment, so you have to go in and turn that repo into a template in the settings.

Incidentally, this particular "fix" means that the new assignment is actually tied to the starter repo from the old version of the assignment, so if the original root assignment that was used to make that first assignment has changed, then those changes will not be reflected in the new assignment you've made ... hope that makes sense.

Not sure why this is so hard for GitHub to fix ...

**Reply by jteno** [reply]
**Thanks for your post.** . Aye this was very confusing. I got this to work by going to the assignment in my old classroom that I wanted to reuse, and changing the assignment type to template. Initially I didn't get any errors when just creating a new classroom and selecting to reuse all the assignments from my old classroom. The new classroom just opened without any assignments.

After setting the assignment that you want to use from and old classroom to "template" type, reusing the assignment in a new classroom works. The new assignment in the new classroom is still linked to the original template source repository where the assignment was originally made from. 

Don't know if this helps or confuses more but:
1. New classroom -> reuse assignment  -> new repo for assignment from old classroom -> that is linked to original source template in original classroom
2. Old classroom -> original assigment that you want to reuse in the new classroom **(you must make this a template!)** -> link to original source template repository.

**Comment by saniales** [comment]
As of today (October 15th, 2025) the error persists and is not fixed

**Comment by jteno** [comment]
I'm running into problems when trying to reuse a group assignment. I think it has to do with the fact that there might not be a group created in the new classroom. As a workaround i tried to create a new group assignment and a group in the classroom as i was suspecting that this was the cause for the problems, but this didn't work.

Has anyone figured out what the problem is with reusing group assignments?

**Comment by menotti** [comment]
As of today (January 23, 2026), the error persists and has not been corrected.

**Comment by nurcanyuruk-2023** [comment]
I had the same issue and was able to fix it by changing the old assignment repository to _template_ as suggested by @khammami.
Detailed steps:
1. Open the old assignment
2. Open the starter code
3. Go to settings
4. Turn on "template repository" feature.

<img width="1893" height="406" alt="Screenshot 2026-02-03 at 11 57 17 AM" src="https://github.com/user-attachments/assets/e19dd8a4-2acd-4e0a-8062-5dcb78409877" />

**Reply by lmbarrero** [reply]
Thanks, it worked!!!

**Comment by fmresearchnovak** [comment]
As of today (Feb. 21st 2026) the issue **_still_** persists.  

When I try to re-use an assignment into a new classroom of the same organization the process simply doesn't work.  No assignment is created in the new classroom.

I think the intended usage is to create a classroom each semester.  I would expect that reusing the assignments from one classroom (say Fall) to the next (say Spring semester) would be the most common use case of the "re-use assignment" feature.  So I am amazed that it doesn't work.
<img width="1187" height="433" alt="Screenshot 2026-02-21 at 7 45 18 AM" src="https://github.com/user-attachments/assets/3d198caa-032a-410d-802e-edf90a10b3c1" />

The feature has worked in the past (+2 years ago) and I did use it.

I attended the SIGCSE Technical Symposium recently which is a very prominent academic conference for computer science educators.  GitHub specifically sends representatives to the conference to promote github classroom and github education in general.  They were not at all prepared to help me with this technical issue and even struggled to recommend what website, webpage, email address or phone number should be used to get technical support.  I am annoyed that there is no mechanism besides a community forum post to report bugs to GitHub employees.  How can the company that has popularized issue tracking for so many projects not have issue tracking for their own product?!

**Reply by fmresearchnovak** [reply]
PS, I am well aware of template repositories and I believe I am using them correctly in this context.

**Reply by jteno** [reply]
Just did my second semester run reusing the assignments.   Enabling templates in the old classrooms original repositories is a workaround but for the group assignments this doesn’t work. Luckily I only have a few. It is a shame how bad the support is. I opened an issue about this and the response time was 6 months and the issue was closed without any proper explanation of what was done.

**Reply by fmresearchnovak** [reply]
Thanks for trying to help!  I agree that the level of official support is not good enough.  We should at least be able to report bugs directly to GitHub employees instead of chatting with other users on a forum.

Can you explain in more detail what you mean by "enabling templates"?  I have been using template repositories to create the assignments in my classrooms for some time now.  Is there something I should be doing differently with template repos?  Please be specific if you can.

Thank you!

**Reply by samus-aran** [reply]
Hi @fmresearchnovak 👋 

Thanks for raising this and for taking the time to share your experience. We really appreciate the feedback, and the team will be in touch shortly via the same means you used to reach us (private email).

For **GitHub Classroom** issues, the best way to get support and tracking is to open an [Education Support ticket](https://support.github.com/contact/education) by selecting the **“GitHub Classroom”** option (as shown in the screenshot below), rather than using _“I have a technical bug”._ This routes your request to the right support queue so it can be triaged and responded to as soon as possible. Hope that clarifies the means to raising a ticket specifically for Classroom.

<img width="649" height="331" alt="Screenshot 2026-02-23 at 19 15 53" src="https://github.com/user-attachments/assets/2520cbfa-f243-4bbe-bdda-498a77f48abf" />

**Reply by bethard** [reply]
@samus-aran Should everyone that has run into this issue file a separate Education Support ticket? Or is there a way to see current Education Support tickets and follow an appropriate one there?

**Reply by samus-aran** [reply]
@bethard Good question. We would ask that you raise your issue individually as its often not always the same but similar BUT when we have a solution suitable for most we'll post in these discussions here where other folks can be guided to or our FAQ.

**Comment by fmresearchnovak** [comment]
I resolved the issue thanks to some detailed information I got from the GitHub support team in a private email.

The issue was that the repository of the old github classroom assignment was not itself a template repo.  I went into the settings of that repo and marked it as a template and **_then_** I was able to reuse that assignment.

An error message on the GitHub website when trying to reuse the assignment would have been really helpful.  Also, the general structure of organizations and template repositories and assignment repositories and student repositories in GitHub classroom is quite confusing and feels unnecessarily complex.

Thanks to the GitHub team for helping me resolve my issue!

**Reply by ssyberg** [reply]
Thanks for this, worked for me!
